docs/guides/pagination/offset-based.md

```diff
@@ -151,198 +126,199 @@ user_data = [
     "age": 20
   }
 ]
+```
 
-@strawberry.type
-class User:
-    name: str = strawberry.field(
-        description="The name of the user."
+Here's the implementation of the `get_pagination_window` function. Note that it is generic and should work for all item types,
+not only for the `User` type.
+
+```py line=99-153
+def get_pagination_window(
+        dataset: List[GenericType],
+        ItemType: type,
+        order_by: str,
+        limit: int | None = None,
+        offset: int = 0,
+        filters: dict[str, str] = {}) -> PaginationWindow:
+    """
+    Get one pagination window on the given dataset for the given limit
+    and offset, ordered by the given attribute and filtered using the
+    given filters
+    """
+
+    # Validate the limit
+    if limit is not None and limit < 1:
+        raise Exception(f'limit ({limit}) must be greater than 0')
+
+    # Sort the dataset
```

Comment:
updated according to your comments.

Reply:
Perhaps I should remove from the documentation the part about Prisma Client Python, since they do not yet support db transactions. What do you prefer?

Reply:
let's remove it, we can have a dedicated guide at some point 😊

Reply:
Done :)